gateways/community/aws-sns/README.md

```diff
@@ -0,0 +1,94 @@
+<p align="center">
+  <img src="https://github.com/argoproj/argo-events/blob/update-docs/docs/assets/sns.png?raw=true" alt="AWS SNS"/>
+</p>
+
+<br/>
+
+# AWS SNS
+
+AWS SNS gateway listens to notifications from SNS.
+
+### How to define an event source in confimap?
+You can construct an entry in configmap using following fields,
+
+```go
+// Hook defines a webhook.
+Hook      *common.Webhook           `json:"hook"`
+
+// TopicArn is the AWS ARN
+TopicArn  string                    `json:"topicArn"`
+
+// K8s secret that holds the access key
+AccessKey *corev1.SecretKeySelector `json:"accessKey" protobuf:"bytes,5,opt,name=accessKey"`
+
+// K8s secret that holds the secret key
+SecretKey *corev1.SecretKeySelector `json:"secretKey" protobuf:"bytes,6,opt,name=secretKey"`
+
+// AWS Region
+Region    string                    `json:"region"`
+```
+
+### Why is there webhook in the gateway?
+Because one of the ways you can receive notifications from SNS is over http. So the gateway runs a http server internally.
+Once you create a new gateway configmap or add a new event source entry in the configmap, the gateway will register the url of the server on AWS.
+All notifications for that topic will then be dispatched by SNS over to the endpoint specified in event source.
+
+### Example
+
+```yaml
```

Comment:
is there anyway to inject this from the examples file directly? that way you remove the necessity of keeping both these sources in sync.. less work maintaining this and more consistency for user experience.

Reply:
I don't that's possible. Maybe the better way is just to reference them instead of actually posting these.